### DISCUSSION TITLE
Power a CC2531 Router via USB?

### DISCUSSION THREAD

**Discussion by Larzeax** [Koenkk/zigbee2mqtt#9614] [Devices]
Hey, 
My Zigbee Network range is a little bit weak, since I moved my Home Assistant to the first floor and upgraded the old Dongle with the new one from Sonoff. Because I had a spear CC2531 with antenna mod lying around I thought I could flash the router firmware on it and use it as an extender. 
I simply wanted to ask if it is possible to power an as router flashed CC2531 with a normal 5W adapter?
And if it is, would you recommend it?

**Comment by francisp2** [comment]
Mine do, no problems with it.

**Comment by AbeltjeNL** [comment]
Yes, this is possible. Did this in the past my self. I stopped using these routers because the firmware is stuck on ZB1.2 and all my other devices are ZB3.0. It works just fine but my network became more stable and responsive without these routers.

**Reply by Larzeax** [reply]
What routers do you use now?
Because I think I broke my CC2531 by accident.

**Reply by AbeltjeNL** [reply]
I only use my mains powered devices as routers. The CC2531 are not in operation anymore. 

How do you mean you broke your router(s)?

**Reply by Larzeax** [reply]
I don't know, I flashed it with the nodemcu method, since I didn't want to wait a few weeks till the debugger would be here. And now the LED is permanently off and the router won't connect to the coordinator.

**Reply by AbeltjeNL** [reply]
Well.. Programming with the debugger is much easier. In the past I only used the debugger method without any problems.

Your stick is probably not broken but has a mis match firmware file. Have you already order a debugger?

**Reply by Larzeax** [reply]
no, I haven't and even if, I had to bend the pins in order to solder the wires to it. But the last time I successfully flashed it as coordinator with a raspberry pi, so I will try to get a zero w or pi 3 4GB and flash it again. I guess the fact, that I'm new to soldering didn't help...
![IMG_0651](https://user-images.githubusercontent.com/53652706/141698555-379b5726-5ce9-4400-a605-e1a89edc4b5d.JPEG)

**Reply by AbeltjeNL** [reply]
Ajajaj.. That's not a pretty face. If you want to you can order these for debuggers from AliExpress. They don't cost a whole lot, the only thing is the waiting time.. 

I'm not familiair with your method of flashing and modding hardware. If there is a decent tutorial and other people managed to do it, you can do it to!
